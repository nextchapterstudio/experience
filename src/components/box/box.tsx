import React, { FC } from 'react'
import classnames from 'classnames'
import { BaseElement, BaseProps } from '../base-element'


const Box : FC<BaseProps>= ({ children, className, ...props }) => {
  return (
    <BaseElement className={classnames('box', className)} {...props}>
      {children}
    </BaseElement>
  )
}



Box.defaultProps = {
  as: 'div',
}

export { Box }

export default Box
