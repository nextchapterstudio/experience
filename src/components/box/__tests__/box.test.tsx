import React from 'react'
import { render, cleanup, fireEvent, screen } from '@testing-library/react'
import { Box } from '../box'

describe('The Box Component', () => {
  afterEach(cleanup)
  it('should render with the class basic', () => {
    const { container } = render(<Box className="basic">test</Box>)

    expect(container).toMatchSnapshot()
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('box')
    expect(container.firstChild).toHaveClass('basic')
  })
})
