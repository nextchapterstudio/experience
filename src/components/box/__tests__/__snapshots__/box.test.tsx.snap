// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Box Component should render with the class basic 1`] = `
<div>
  <div
    class="box basic"
  >
    test
  </div>
</div>
`;

exports[`The Box Component should render with the class basic 2`] = `
<div
  class="box basic"
>
  test
</div>
`;
